lib/osv/ecosystems.py

```diff
@@ -327,23 +332,36 @@ def __init__(self, debian_release_ver: str):
   def sort_key(self, version):
     return DebianVersion.from_string(version)
 
-  def enumerate_versions(self, package, introduced, fixed, limits=None):
+  def _get_package_version_response(self, package):
+    """Return the response from debian snapshot api for the given package"""
     url = self._API_PACKAGE_URL.format(package=package.lower())
+
+    if redis_cache is not None:
```

Comment:
Let's factor this all out into a reusable `RequestHelper` class (or something similarly named).

```python
class RequestHelper:
  retry_total: int = 0
  backoff_factor: int = 0
  cache: Cache = None
  cache_ttl:  int = 0

  def get(url):
    ...
```

Then we can just use this like:

```python
request_helper = RequestHelper(
  self.RETRY_TOTAL, self._BACKOFF_FACTOR, 
  cache=global_cache,
  cache_ttl=self._CACHE_TTL)

response = request_helper.get(...)
```